opt: panic: buildMultiRowSubquery called with operator >=

The only thing I can do is profusely apologize for the utter ridiculousness of this query and how long it will take to figure out where the problem is.

```
CREATE TABLE t (i INT8, f FLOAT8, d DECIMAL, s STRING, b BYTES, z BOOL);

WITH
    with_954 (col_11709, col_11710, col_11711, col_11712, col_11713, col_11714, col_11715, col_11716, col_11717, col_11718, col_11719, col_11720, col_11721)
        AS (
            SELECT
                *
            FROM
                (
                    VALUES
                        (
                            '1970-01-10 04:26:04.000618+00:00':::TIMESTAMPTZ,
                            '45.51.229.206/31':::INET,
                            3553678228718548667:::INT8::INT8
                            + (
                                    WITH
                                        with_951 (col_11697, col_11698, col_11699)
                                            AS (
                                                SELECT
                                                    tab_4652.f AS col_11697, '1973-05-08':::DATE AS col_11698, '2ac982bb-fb19-4adc-b81c-b277ca6e2885':::UUID AS col_11699
                                                FROM
                                                    defaultdb.public.t AS tab_4651
                                                    INNER JOIN defaultdb.public.t AS tab_4652
                                                    ON
                                                        EXISTS(
                                                            WITH
                                                                with_950 (col_11670, col_11671, col_11672, col_11673)
                                                                    AS (
                                                                        SELECT
                                                                            *
                                                                        FROM
                                                                            (
                                                                                VALUES
                                                                                    (
                                                                                        CASE
                                                                                        WHEN inet_contains_or_equals(
                                                                                            (
                                                                                                '52d4:ac5e:1aa8:40ad:c4f8:7f0b:baff:53a6/45':::INET::INET - (-8066856872739691600):::INT8::INT8
                                                                                            )::INET,
                                                                                            '33.108.211.28/15':::INET::INET
                                                                                        )::BOOL
                                                                                        THEN CASE WHEN true THEN 3993606345:::OID ELSE 3835004652:::OID END
                                                                                        ELSE 648866301:::OID
                                                                                        END,
                                                                                        NULL,
                                                                                        B'000100110100111111000000101011',
                                                                                        (-4642574565784178043):::INT8
                                                                                    ),
                                                                                    (
                                                                                        2942858690:::OID,
                                                                                        (-434694773324.28031):::DECIMAL,
                                                                                        B'11101000010100000010100000001110010',
                                                                                        array_position(
                                                                                            ARRAY[e'\\xef489ccda9e6f9cfb2':::BYTES]::BYTES[],
                                                                                            convert_to(
                                                                                                'm':::STRING::STRING,
                                                                                                (e'#\x13\x122':::STRING::STRING || e'O<OB[\x12\x04\x14x':::STRING::STRING)::STRING
                                                                                            )::BYTES::BYTES
                                                                                        )::INT8
                                                                                    ),
                                                                                    (
                                                                                        3398031786:::OID,
                                                                                        NULL,
                                                                                        (
                                                                                            WITH
                                                                                                with_948
                                                                                                    (
                                                                                                        col_11654,
                                                                                                        col_11655,
                                                                                                        col_11656,
                                                                                                        col_11657,
                                                                                                        col_11658,
                                                                                                        col_11659,
                                                                                                        col_11660,
                                                                                                        col_11661,
                                                                                                        col_11662,
                                                                                                        col_11663,
                                                                                                        col_11664,
                                                                                                        col_11665
                                                                                                    )
                                                                                                    AS (
                                                                                                        SELECT
                                                                                                            ret_1646::VARBIT # ret_1646::VARBIT AS col_11654,
                                                                                                            ret_1652 AS col_11655,
                                                                                                            ret_1650 AS col_11656,
                                                                                                            ret_1656 AS col_11657,
                                                                                                            CASE WHEN ret_1647 THEN ret_1654 ELSE ret_1654 END AS col_11658,
                                                                                                            jsonb_insert(
                                                                                                                'false':::JSONB::JSONB,
                                                                                                                ARRAY[]:::STRING[]::STRING[],
                                                                                                                '[[[{"c": {}}, []]], {}, [{}, {}], {"N!_": true}, {}]':::JSONB::JSONB
                                                                                                            )::JSONB::JSONB->>e'\x022J4\x13':::STRING::STRING
                                                                                                                AS col_11659,
                                                                                                            ret_1648 AS col_11660,
                                                                                                            ret_1641 AS col_11661,
                                                                                                            ret_1647 AS col_11662,
                                                                                                            ret_1646 AS col_11663,
                                                                                                            ret_1646 AS col_11664,
                                                                                                            e'[{"$Wb#O|b": false, "MuMlwb9|ck&g": true, "p`(b8K)/": [[], false, false]}, {"e8\\\\]}fh(+t": false}, []]':::JSONB
                                                                                                                AS col_11665
                                                                                                        FROM
                                                                                                            [
                                                                                                                DELETE FROM
                                                                                                                    defaultdb.public.t AS tab_4653
                                                                                                                WHERE
                                                                                                                    tab_4653.z
                                                                                                                RETURNING
                                                                                                                    CASE
                                                                                                                    WHEN (
                                                                                                                        (
                                                                                                                            z
                                                                                                                            OR isnan(f::FLOAT8)::BOOL
                                                                                                                                >= EXISTS(
                                                                                                                                        SELECT
                                                                                                                                            '1970-01-11 22:14:37.000599+00:00':::TIMESTAMP::TIMESTAMP
                                                                                                                                            - '-21 years -11 mons -52 days -10:55:56.270713':::INTERVAL::INTERVAL
                                                                                                                                                AS col_11653
                                                                                                                                        FROM
                                                                                                                                            defaultdb.public.t AS tab_4654
                                                                                                                                        ORDER BY
                                                                                                                                            tab_4654.b ASC
                                                                                                                                        LIMIT
                                                                                                                                            28:::INT8
                                                                                                                                    )
                                                                                                                        )
                                                                                                                        OR z
                                                                                                                    )
                                                                                                                    THEN b
                                                                                                                    ELSE e'\\xa270ff2148':::BYTES
                                                                                                                    END
                                                                                                                        AS ret_1641,
                                                                                                                    '1970-01-07 13:52:46.000081+00:00':::TIMESTAMPTZ AS ret_1642,
                                                                                                                    i AS ret_1643,
                                                                                                                    '255.209.215.69/27':::INET AS ret_1644,
                                                                                                                    '-51 years -6 mons -118 days -23:02:47.621194':::INTERVAL AS ret_1645,
                                                                                                                    B'0111100001000011010001111101111100100000000001010000010110100000001010011110111'
                                                                                                                        AS ret_1646,
                                                                                                                    z AS ret_1647,
                                                                                                                    '10:23:14.214796':::TIME AS ret_1648,
                                                                                                                    '1.83.227.207/16':::INET::INET | '224.65.31.89/26':::INET::INET
                                                                                                                        AS ret_1649,
                                                                                                                    f AS ret_1650,
                                                                                                                    i AS ret_1651,
                                                                                                                    f AS ret_1652,
                                                                                                                    uuid_v4()::BYTES AS ret_1653,
                                                                                                                    NULL AS ret_1654,
                                                                                                                    CASE WHEN z THEN sqrt(f::FLOAT8)::FLOAT8 ELSE f END AS ret_1655,
                                                                                                                    '6fb1d128-9cc9-45ba-bf8d-d9ae7bcd2451':::UUID AS ret_1656
                                                                                                            ]
                                                                                                        WHERE
                                                                                                            ret_1647
                                                                                                    ),
                                                                                                with_949 (col_11666, col_11667, col_11668)
                                                                                                    AS (
                                                                                                        SELECT
                                                                                                            *
                                                                                                        FROM
                                                                                                            (
                                                                                                                VALUES
                                                                                                                    (
                                                                                                                        883982675:::OID,
                                                                                                                        0.4286330666549404:::FLOAT8,
                                                                                                                        '30 years 2 mons 576 days 23:15:18.811011':::INTERVAL
                                                                                                                    ),
                                                                                                                    (
                                                                                                                        945883331:::OID,
                                                                                                                        (-0.9908750694471347):::FLOAT8,
                                                                                                                        '-76 years -11 mons -637 days -22:32:22.989716':::INTERVAL
                                                                                                                    ),
                                                                                                                    (2082337326:::OID, 1.2328627206822742:::FLOAT8, NULL),
                                                                                                                    (
                                                                                                                        1047652034:::OID,
                                                                                                                        (-0.8909456956582038):::FLOAT8,
                                                                                                                        '-29 years -11 mons -198 days -19:33:02.001645':::INTERVAL
                                                                                                                    ),
                                                                                                                    (NULL, NULL, '-20 years -6 mons -465 days -07:22:38.092363':::INTERVAL)
                                                                                                            )
                                                                                                                AS tab_4655 (col_11666, col_11667, col_11668)
                                                                                                    )
                                                                                            SELECT
                                                                                                B'11001101111111111011111101001100010001110011001' AS col_11669
                                                                                            FROM
                                                                                                with_949
                                                                                            WHERE
                                                                                                NULL
                                                                                            ORDER BY
                                                                                                with_949.col_11667 ASC, with_949.col_11668 ASC, with_949.col_11667 DESC
                                                                                            LIMIT
                                                                                                1:::INT8
                                                                                        ),
                                                                                        (-1837590048758234679):::INT8
                                                                                    ),
                                                                                    (
                                                                                        2683321701:::OID,
                                                                                        NULL,
                                                                                        B'1011000100010011111000111101001011000111010010000100100001001110011',
                                                                                        5868345694159578265:::INT8
                                                                                    ),
                                                                                    (11769654:::OID, 1.794454392804474730E+32:::DECIMAL, B'10111011100011', 4400576352155193947:::INT8)
                                                                            )
                                                                                AS tab_4656 (col_11670, col_11671, col_11672, col_11673)
                                                                    )
                                                            SELECT
                                                                NULL AS col_11674,
                                                                B'110010110001000110110101111010000001100010111101011001110' AS col_11675,
                                                                '1970-01-04 00:07:01.000559+00:00':::TIMESTAMP AS col_11676,
                                                                '1970-01-08 22:53:42.000338+00:00':::TIMESTAMPTZ AS col_11677,
                                                                tab_4657.i AS col_11678,
                                                                tab_4657.z AS col_11679,
                                                                '7dbe:5be1:52bd:8390:4242:8981:ccf4:d221/2':::INET AS col_11680,
                                                                tab_4657.s AS col_11681,
                                                                '1970-01-08 11:16:47.000125+00:00':::TIMESTAMP AS col_11682,
                                                                tab_4657.z AS col_11683,
                                                                tab_4657.b AS col_11684,
                                                                '[[[[{"p!L6}}2g": true}], false, {}, [false, [[]], 0.5073787025280246]]], {"} cJO8Y7zj": []}, {}, []]':::JSONB AS col_11685,
                                                                '72 years 6 mons 297 days 04:15:59.164365':::INTERVAL AS col_11686,
                                                                'e75620b8-5249-4515-ba3a-7f0cab49c855':::UUID AS col_11687,
                                                                tab_4657.i AS col_11688,
                                                                '-80 years -7 mons -764 days -16:52:45.112635':::INTERVAL AS col_11689,
                                                                3597295236:::OID AS col_11690,
                                                                tab_4657.f AS col_11691,
                                                                tab_4657.b AS col_11692,
                                                                tab_4657.z AS col_11693,
                                                                '1979-08-07':::DATE AS col_11694,
                                                                tab_4657.s AS col_11695,
                                                                '-83 years -3 mons -2 days -22:18:06.994551':::INTERVAL AS col_11696
                                                            FROM
                                                                defaultdb.public.t AS tab_4657
                                                            LIMIT
                                                                34:::INT8
                                                        )
                                            ),
                                        with_952 (col_11700, col_11701)
                                            AS (
                                                SELECT
                                                    *
                                                FROM
                                                    (
                                                        VALUES
                                                            (
                                                                '{"-wQ9xWh)jP": [], "TP>-T": {"a": [[]], "foo": [false]}, "b": [], "bar": [null], "baz": {}, "c": [null]}':::JSONB,
                                                                '{" if&#ZT/x>$<": {}, "b": {"6Vr4G": {"`,b": false}, "7DofqLc;}": [{}, [null]]}, "foo": {"OZ8(q<o9;A": 0.12940848092770751}, "mscuoc~VWg": [0.5640764347768751, {"b": {"ee<k2U,Zq": []}}]}':::JSONB
                                                            ),
                                                            (
                                                                e'[{"foo": 2.419101329737345, "foobar": {")]Akx(": false, ";X=<W!/1gb]i": true, "S]ne\\\\\']": [[]], "foo": {")BDWZpcM#_%": false, "[@qeI8g4eq<&": false}, "li*Il": {}}}, {"a": []}, null, []]':::JSONB,
                                                                e'{"!/[` e7&hSbE": [true], "WW<4X{;6cJJ7": [{"Qq5[Dk\\"J": {"Xujz": "c", "b": ["foo"]}}, {"osO0`LFoH": {">ka": [], "a": [[]]}}, [], false], "foo": {"yj|wG": []}, "w:4\\"nmR": []}':::JSONB
                                                            ),
                                                            (
                                                                '[{"Z0ac;n~ZI`": [], "b": [], "foobar": {}, "kpjp%&GG8;s": "Ot=$o$("}, [[], {"]o#sw1{]bA": true}, {}], []]':::JSONB,
                                                                e'[{"bar": {"b": []}, "c": {"(OiShNKG>p=": {}, "C%\\"|m": []}}, 0.017990701863495016, {"NS1KlV)-F": []}, []]':::JSONB
                                                            ),
                                                            (NULL, e'{"\\"e~u\\\\/": [], "PMGO": [], "a": [], "baz": "a", "foobar": false}':::JSONB)
                                                    )
                                                        AS tab_4658 (col_11700, col_11701)
                                            ),
                                        with_953 (col_11702, col_11703, col_11704, col_11705, col_11706, col_11707)
                                            AS (
                                                SELECT
                                                    *
                                                FROM
                                                    (
                                                        VALUES
                                                            (
                                                                NULL,
                                                                'aa8b45b1-f27d-4839-80bc-04c0efa0ab6f':::UUID,
                                                                e'eC[\x0b\r^':::STRING,
                                                                '1970-01-09 23:26:31.000072+00:00':::TIMESTAMPTZ,
                                                                '1970-01-05 09:57:56.000879+00:00':::TIMESTAMPTZ,
                                                                NULL
                                                            )
                                                    )
                                                        AS tab_4659 (col_11702, col_11703, col_11704, col_11705, col_11706, col_11707)
                                            )
                                    SELECT
                                        '1993-11-10':::DATE AS col_11708
                                    FROM
                                        defaultdb.public.t AS tab_4660, defaultdb.public.t AS tab_4661
                                    WHERE
                                        tab_4660.z
                                    LIMIT
                                        1:::INT8
                                )::DATE,
                            NULL,
                            '1398:5c93:bfe7:c7fe:ed41:5fae:9a96:80e7/100':::INET,
                            '76159b28-73fb-493b-8045-e62d4560219a':::UUID,
                            NULL,
                            '10:37:03.861632':::TIME,
                            e'{"baz": {"a": {"c": []}, "fE>": [{"b": [], "foobar": "ztvJ}$[-"}, [[true, [0.39402288114852335]], {}], {}], "foobar": {"c": false, "w:\\"x?0B": {}}}}':::JSONB,
                            2067549112:::OID,
                            true,
                            NULL,
                            1117639022:::OID
                        ),
                        (
                            '1970-01-09 09:04:19.000531+00:00':::TIMESTAMPTZ,
                            '197.217.123.243/2':::INET,
                            NULL,
                            '1975-12-09':::DATE,
                            '41.219.155.89/11':::INET,
                            '9a98d711-c9e2-41d1-ab50-a0dd9d79d807':::UUID,
                            NULL,
                            NULL,
                            '[{"*;zN2,i[_": [{}], "baz": [true]}, 1.1352958506878519]':::JSONB,
                            3121423325:::OID,
                            true,
                            '1970-01-04 00:50:59.000834+00:00':::TIMESTAMPTZ,
                            129888878:::OID
                        )
                )
                    AS tab_4662 (col_11709, col_11710, col_11711, col_11712, col_11713, col_11714, col_11715, col_11716, col_11717, col_11718, col_11719, col_11720, col_11721)
        ),
    with_955 (col_11722, col_11723, col_11724)
        AS (
            SELECT
                *
            FROM
                (VALUES ('04:22:18.59264':::TIME, NULL, B'10111110111010011010010110011011110111111110101111011011111110111000111110110110111')) AS tab_4663 (col_11722, col_11723, col_11724)
        )
SELECT
    e'[{">l2_}sF\\\\>-": true, "bar": {"__V\\\\ix\\"": null, "a": 0.25364007462656823}, "baz": {}, "gj_>zepILKjj": {"FT$da": {"bar": false}, "c": [0.28093873887935006, [], true, {}]}, "nrElipF\\\\{}J": {"U4/wJ%\\\\+": {"P d": "b"}}}, []]':::JSONB
        AS col_11725,
    tab_4664.d AS col_11726,
    tab_4664.b AS col_11727,
    NULL AS col_11728
FROM
    defaultdb.public.t AS tab_4664, with_955
WHERE
    tab_4664.z
ORDER BY
    tab_4664.b ASC, tab_4664.b DESC
LIMIT
    36:::INT8;
```

```
panic: buildMultiRowSubquery called with operator >= [recovered]
	panic: buildMultiRowSubquery called with operator >= [recovered]
	panic: panic while executing 1 statements: WITH _ (_, _, _, _, _, _, _, _, _, _, _, _, _) AS (SELECT * FROM (VALUES (_:::TIMESTAMPTZ, _:::INET, (_:::INT8::INT8 + (WITH _ (_, _, _) AS (SELECT _._ AS _, _:::DATE AS _, _:::UUID AS _ FROM _._._ AS _ INNER JOIN _._._ AS _ ON EXISTS (WITH _ (_, _, _, _) AS (SELECT * FROM (VALUES (CASE WHEN inet_contains_or_equals((_::INET - (_):::INT8::INT8)::INET, _::INET)::BOOL THEN CASE WHEN _ THEN _ ELSE _ END ELSE _:::OID END, _, _, (_):::INT8), (__more4__)) AS _ (_, _, _, _)) SELECT _ AS _, _ AS _, _:::TIMESTAMP AS _, _:::TIMESTAMPTZ AS _, _._ AS _, _._ AS _, _:::INET AS _, _._ AS _, _:::TIMESTAMP AS _, _._ AS _, _._ AS _, _:::JSONB AS _, _:::INTERVAL AS _, _:::UUID AS _, _._ AS _, _:::INTERVAL AS _, _:::OID AS _, _._ AS _, _._ AS _, _._ AS _, _:::DATE AS _, _._ AS _, _:::INTERVAL AS _ FROM _._._ AS _ LIMIT _:::INT8)) , _ (_, _) AS (SELECT * FROM (VALUES (_:::JSONB, _:::JSONB), (__more3__)) AS _ (_, _)) , _ (_, _, _, _, _, _) AS (SELECT * FROM (VALUES (_, _:::UUID, _:::STRING, _:::TIMESTAMPTZ, _:::TIMESTAMPTZ, _)) AS _ (_, _, _, _, _, _)) SELECT _:::DATE AS _ FROM _._._ AS _, _._._ AS _ WHERE _._ LIMIT _:::INT8)::DATE), _, _:::INET, _:::UUID, _, _:::TIME, _:::JSONB, _:::OID, _, _, _:::OID), (__more1__)) AS _ (_, _, _, _, _, _, _, _, _, _, _, _, _)) , _ (_, _, _) AS (SELECT * FROM (VALUES (_:::TIME, _, _)) AS _ (_, _, _)) SELECT _:::JSONB AS _, _._ AS _, _._ AS _, _ AS _ FROM _._._ AS _, _ WHERE _._ ORDER BY _._ ASC, _._ DESC LIMIT _:::INT8; caused by buildMultiRowSubquery called with operator >=

goroutine 140 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00180db00, 0x2f10580, 0xc00152e580, 0x2536380, 0xc0002fcc50)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:715 +0x36d
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc00180db00, 0x2f10580, 0xc00152e580)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:427 +0x61
panic(0x2536380, 0xc0002fcc50)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build.func1(0xc003094858)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:142 +0x9a
panic(0x2536380, 0xc0002fcc50)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildMultiRowSubquery(0xc00099b3b0, 0xc001a81860, 0xc001dcfde0, 0x0, 0x0, 0x0, 0x2f2ca60)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/subquery.go:286 +0x4e1
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildScalar(0xc00099b3b0, 0x2f253e0, 0xc001a81860, 0xc001dcfde0, 0x0, 0x0, 0x0, 0x2f2ca60, 0xc0018b22a0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:248 +0x24b3
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildScalar(0xc00099b3b0, 0x2f25ec0, 0xc0016a14d0, 0xc001dcfde0, 0x0, 0x0, 0x0, 0x1, 0x2)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:298 +0x1502
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildScalar(0xc00099b3b0, 0x2f25ec0, 0xc0016a1500, 0xc001dcfde0, 0x0, 0x0, 0x0, 0x2847fa0, 0x4866000)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:297 +0x1477
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildScalar(0xc00099b3b0, 0x2f251a0, 0xc001a818b0, 0xc001dcfde0, 0xc001dcfed0, 0xc001ade900, 0x0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scalar.go:215 +0x2b86
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildProjectionList(0xc00099b3b0, 0xc001dcfde0, 0xc001dcfed0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/project.go:167 +0x68
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*mutationBuilder).buildReturning(0xc003090d78, 0xc001ef1400, 0x10, 0x10)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/mutation_builder.go:530 +0x564
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*mutationBuilder).buildDelete(0xc003090d78, 0xc001ef1400, 0x10, 0x10)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/delete.go:92 +0xce
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDelete(0xc00099b3b0, 0xc0011405f0, 0xc001dcf2d0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/delete.go:78 +0x32b
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f15240, 0xc0011405f0, 0xc001dcf2d0, 0xc00025af80)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:207 +0x1e6
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDataSource(0xc00099b3b0, 0x2ef2580, 0xc0002508c0, 0x0, 0xc001dcf2d0, 0x1618ac2)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:121 +0xae9
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDataSource(0xc00099b3b0, 0x2ef2040, 0xc001140640, 0x0, 0xc001dcf2d0, 0xc004300510)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:57 +0x2db
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFromTables(0xc00099b3b0, 0xc0002508d0, 0x1, 0x1, 0xc001dcf2d0, 0xc001dcf1e0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:743 +0x75
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFrom(0xc00099b3b0, 0xc001ca5e00, 0xc001dcf2d0, 0xffca77)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:689 +0x2b3
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelectClause(0xc00099b3b0, 0xc0019bbe60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc001dcf2d0, 0xc001f608e8)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:630 +0x5e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc00163d900, 0x0, 0x0, 0x0, 0xc001dcf2d0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:585 +0x201
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f15d00, 0xc00163d900, 0xc001dcf2d0, 0x2362)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:219 +0x10e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildCTE(0xc00099b3b0, 0xc000251150, 0x2, 0x2, 0xc001dcf1e0, 0x2f28080)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:448 +0x390
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc00163de80, 0x0, 0x0, 0x0, 0xc001dcf1e0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:578 +0x7a5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f158c0, 0xc0033db040, 0xc001dcf1e0, 0xc0033db040)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:216 +0x2de
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).replaceSubquery(0xc001dcf1e0, 0xc001f2a3f0, 0xc001f60800, 0x1, 0xc001c6bc00, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1099 +0x1ac
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre(0xc001dcf1e0, 0x2f165c0, 0xc001f2a3f0, 0x2f27a01, 0x48660c0, 0x2f277e0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:875 +0x130
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr(0x2ef1a40, 0xc001dcf1e0, 0x2f165c0, 0xc001f2a3f0, 0x4866790, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:680 +0x56
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).walkExprTree(0xc001dcf1e0, 0x2f165c0, 0xc001f2a3f0, 0x48660c0, 0x4866001)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:263 +0x5a
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).resolveType(0xc001dcf1e0, 0x2f165c0, 0xc001f2a3f0, 0x2f27a80, 0x48660c0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:299 +0x43
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildValuesClause(0xc00099b3b0, 0xc004300320, 0x0, 0x0, 0x0, 0xc001dcf1e0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/values.go:66 +0x2e3
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc00163df80, 0x0, 0x0, 0x0, 0xc001dcf1e0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:591 +0x6cc
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f158c0, 0xc0033db048, 0xc001dcf1e0, 0xc0033db048)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:216 +0x2de
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDataSource(0xc00099b3b0, 0x2ef25c0, 0xc001f2a720, 0x0, 0xc001dcf1e0, 0x8)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:110 +0x174
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDataSource(0xc00099b3b0, 0x2ef2040, 0xc001140a00, 0x0, 0xc001dcf1e0, 0x203000)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:57 +0x2db
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFromTables(0xc00099b3b0, 0xc0002513c0, 0x1, 0x1, 0xc001dcf1e0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:743 +0x75
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFrom(0xc00099b3b0, 0xc001f2a750, 0xc001dcf1e0, 0xc003092478)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:689 +0x2b3
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelectClause(0xc00099b3b0, 0xc0009b01b0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc001dcf1e0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:630 +0x5e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc001fbe000, 0x0, 0x0, 0x0, 0xc001dcf1e0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:585 +0x201
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f15d00, 0xc001fbe000, 0xc001dcf1e0, 0x15dddfc)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:219 +0x10e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildCTE(0xc00099b3b0, 0xc0033db050, 0x1, 0x1, 0xc001dcf0f0, 0xc003092878)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:448 +0x390
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc001fbe640, 0x0, 0x0, 0x0, 0xc001dcf0f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:578 +0x7a5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f158c0, 0xc0033db058, 0xc001dcf0f0, 0xc0033db058)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:216 +0x2de
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).replaceSubquery(0xc001dcf0f0, 0xc001f2aa20, 0x1, 0xffffffffffffffff, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1099 +0x1ac
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre(0xc001dcf0f0, 0x2f165c0, 0xc001f2aa20, 0x7, 0x1, 0xc003092b78)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:873 +0xc5
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr(0x2ef1a40, 0xc001dcf0f0, 0x2f165c0, 0xc001f2aa20, 0xc0012e5000, 0x7, 0xc003092bf8)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:680 +0x56
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).walkExprTree(0xc001dcf0f0, 0x2f165c0, 0xc001f2aa20, 0xc001dcf400, 0x7)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:263 +0x5a
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).resolveAndRequireType(0xc001dcf0f0, 0x2f165c0, 0xc001f2aa20, 0x2f27b40, 0x48660c0, 0xc001dcf000, 0xc00118ebd0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:318 +0x49
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildJoin(0xc00099b3b0, 0xc001954be0, 0xc001dce9e0, 0xc000f64480)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/join.go:95 +0x5f0
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDataSource(0xc00099b3b0, 0x2ef2400, 0xc001954be0, 0x0, 0xc001dce9e0, 0x1bdfb69)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:69 +0x918
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFromTables(0xc00099b3b0, 0xc0002fc090, 0x1, 0x1, 0xc001dce9e0, 0xc003093090)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:743 +0x75
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFrom(0xc00099b3b0, 0xc001f2aa50, 0xc001dce9e0, 0xc0011fbed0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:689 +0x2b3
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelectClause(0xc00099b3b0, 0xc0009b0630, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc001dce9e0, 0x10)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:630 +0x5e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc001fbe680, 0x0, 0x0, 0x0, 0xc001dce9e0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:585 +0x201
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f15d00, 0xc001fbe680, 0xc001dce9e0, 0x10)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:219 +0x10e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildCTE(0xc00099b3b0, 0xc001c6aa60, 0x3, 0x4, 0xc001dce8f0, 0x10)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:448 +0x390
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc001fbf2c0, 0x0, 0x0, 0x0, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:578 +0x7a5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f158c0, 0xc0033db078, 0xc001dce8f0, 0xc0033db078)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:216 +0x2de
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).replaceSubquery(0xc001dce8f0, 0xc001f2af30, 0xc003093700, 0xffffffffffffffff, 0xc001dce800, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:1099 +0x1ac
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).VisitPre(0xc001dce8f0, 0x2f165c0, 0xc001f2af30, 0x2f14580, 0xc001ca5080, 0xc001ca5080)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:875 +0x130
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr(0x2ef1a40, 0xc001dce8f0, 0x2f165c0, 0xc001f2af30, 0x2f14580, 0xc001ca5080, 0xc000f06700)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:680 +0x56
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*CastExpr).Walk(0xc001fbf5c0, 0x2ef1a40, 0xc001dce8f0, 0x2f14801, 0x2f14800)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:131 +0x4f
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr(0x2ef1a40, 0xc001dce8f0, 0x2f14800, 0xc001fbf5c0, 0x2f14800, 0xc0006e6f00, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:683 +0x227
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*BinaryExpr).Walk(0xc001fbf900, 0x2ef1a40, 0xc001dce8f0, 0x2f27d01, 0x2f146c0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:73 +0xb1
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr(0x2ef1a40, 0xc001dce8f0, 0x2f146c0, 0xc001fbf900, 0x2f16580, 0xc0006e6ec0, 0x2f27d20)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:683 +0x227
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*ParenExpr).Walk(0xc001c6ac00, 0x2ef1a40, 0xc001dce8f0, 0x2f27d01, 0x2f15880)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:466 +0x4f
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.WalkExpr(0x2ef1a40, 0xc001dce8f0, 0x2f15880, 0xc001c6ac00, 0xc000f06740, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/walk.go:683 +0x227
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).walkExprTree(0xc001dce8f0, 0x2f15880, 0xc001c6ac00, 0x48660c0, 0x4866001)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:263 +0x5a
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*scope).resolveType(0xc001dce8f0, 0x2f15880, 0xc001c6ac00, 0x2f27a80, 0x48660c0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/scope.go:299 +0x43
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildValuesClause(0xc00099b3b0, 0xc001c6ad80, 0x0, 0x0, 0x0, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/values.go:66 +0x2e3
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc001254880, 0x0, 0x0, 0x0, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:591 +0x6cc
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f158c0, 0xc0033db080, 0xc001dce8f0, 0xc0033db080)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:216 +0x2de
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDataSource(0xc00099b3b0, 0x2ef25c0, 0xc001f2b710, 0x0, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:110 +0x174
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildDataSource(0xc00099b3b0, 0x2ef2040, 0xc0014e4320, 0x0, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:57 +0x2db
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFromTables(0xc00099b3b0, 0xc0002fc3e0, 0x1, 0x1, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:743 +0x75
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildFrom(0xc00099b3b0, 0xc001f2b740, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:689 +0x2b3
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelectClause(0xc00099b3b0, 0xc0009b0990, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc001dce8f0, 0xc002f02ab8)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:630 +0x5e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc001254980, 0x0, 0x0, 0x0, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:585 +0x201
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f15d00, 0xc001254980, 0xc001dce8f0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:219 +0x10e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildCTE(0xc00099b3b0, 0xc0002fc6f0, 0x2, 0x2, 0xc001dce800, 0xc001254f00)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:448 +0x390
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildSelect(0xc00099b3b0, 0xc001254e40, 0x0, 0x0, 0x0, 0xc001dce800, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/select.go:578 +0x7a5
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).buildStmt(0xc00099b3b0, 0x2f15d00, 0xc001254e40, 0xc001dce800, 0x4)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:219 +0x10e
github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder.(*Builder).Build(0xc00099b3b0, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/opt/optbuilder/builder.go:156 +0x111
github.com/cockroachdb/cockroach/pkg/sql.(*optPlanningCtx).buildExecMemo(0xc00180e5b0, 0x2f10640, 0xc001f2ba10, 0x0, 0xc00ba3, 0x2f10640, 0xc001f2ba10)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:421 +0x18e
github.com/cockroachdb/cockroach/pkg/sql.(*planner).makeOptimizerPlan(0xc00180de50, 0x2f10640, 0xc001f2ba10, 0x289c958, 0x19, 0x1893e94e7afbb, 0x2e1efb4d)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/plan_opt.go:155 +0xa2
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).makeExecPlan(0xc00180db00, 0x2f10640, 0xc001f2ba10, 0xc00180de50, 0x6, 0x5f072f)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:979 +0x131
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00180db00, 0x2f10640, 0xc001f2ba10, 0xc00180de50, 0x7f150d5dcc68, 0xc000f25290, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:866 +0x148
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00180db00, 0x2f10640, 0xc001f2ba10, 0x2f15d00, 0xc001254e40, 0xc001118f00, 0x2354, 0x0, 0x0, 0x0, ...)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:457 +0xdd4
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00180db00, 0x2f10640, 0xc001f2ba10, 0x2f15d00, 0xc001254e40, 0xc001118f00, 0x2354, 0x0, 0x0, 0x0, ...)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:103 +0x610
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00180db00, 0x2f10580, 0xc00152e580, 0xc000122e78, 0x5400, 0x15000, 0xc000122f10, 0xc0002fd290, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1178 +0x21bb
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00053a340, 0x2f10580, 0xc00152e580, 0xc00180db00, 0x5400, 0x15000, 0xc000122f10, 0xc0002fd290, 0x0, 0x0)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:429 +0xce
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl.func4(0xc00053a340, 0xc000c2c000, 0xc000367034, 0xc0002fd290, 0x2f10580, 0xc00152e580, 0xc00180db00, 0x5400, 0x15000, 0xc000122f10, ...)
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:332 +0xcf
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).serveImpl
	/home/mjibson/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:315 +0x1040
```

Found with sqlsmith. Obviously.

--- a/pkg/sql/opt/optbuilder/scalar.go
+++ b/pkg/sql/opt/optbuilder/scalar.go
@@ -244,14 +244,14 @@ func (b *Builder) buildScalar(
 		out = b.factory.ConstructColumnAccess(input, memo.TupleOrdinal(t.ColIndex))
 
 	case *tree.ComparisonExpr:
-		if sub, ok := t.Right.(*subquery); ok && sub.wrapInTuple {
+		if sub, ok := t.Right.(*subquery); ok && sub.isMultiRow() {
 			out, _ = b.buildMultiRowSubquery(t, inScope, colRefs)
 			// Perform correctness checks on the outer cols, update colRefs and
 			// b.subquery.outerCols.
 			b.checkSubqueryOuterCols(sub.outerCols, inGroupingContext, inScope, colRefs)
 		} else if b.hasSubOperator(t) {
-			// Cases where the RHS is a subquery and not a scalar (of which only an
-			// array or tuple is legal) were handled above.
+			// Cases where the RHS is a multi-row subquery were handled above, so this
+			// only handles explicit tuples and arrays.
 			out = b.buildAnyScalar(t, inScope, colRefs)
 		} else {
 			left := b.buildScalar(t.TypedLeft(), inScope, nil, nil, colRefs)

--- a/pkg/sql/opt/optbuilder/subquery.go
+++ b/pkg/sql/opt/optbuilder/subquery.go
@@ -55,6 +55,11 @@ type subquery struct {
 	outerCols opt.ColSet
 }
 
+// isMultiRow returns whether the subquery can return multiple rows.
+func (s *subquery) isMultiRow() bool {
+	return s.wrapInTuple && !s.Exists
+}
+
 // Walk is part of the tree.Expr interface.
 func (s *subquery) Walk(v tree.Visitor) tree.Expr {
 	return s

--- a/pkg/sql/opt/optbuilder/testdata/subquery
+++ b/pkg/sql/opt/optbuilder/testdata/subquery
@@ -742,6 +742,24 @@ project
                      └── columns: a:1(int!null) b:2(int) c:3(int)
 
 build
+SELECT true = EXISTS (SELECT 1)
+----
+project
+ ├── columns: "?column?":2(bool)
+ ├── values
+ │    └── tuple [type=tuple]
+ └── projections
+      └── eq [type=bool]
+           ├── true [type=bool]
+           └── exists [type=bool]
+                └── project
+                     ├── columns: "?column?":1(int!null)
+                     ├── values
+                     │    └── tuple [type=tuple]
+                     └── projections
+                          └── const: 1 [type=int]
+
+build
 SELECT (SELECT a FROM abc WHERE false)
 ----
 project

